app/js/lib/foxbox/api.js

```diff
@@ -148,4 +222,78 @@ export default class API {
 
     return new Promise((resolve) => settings.once('session', () => resolve()));
   }
+
+  /**
+   * Fetches values for the set of getters.
+   *
+   * @return {Promise}
+   * @private
+   */
+  [p.fetchGetterValues]() {
+    // It may happen that all watchers have been unregistered in the meantime,
+    // so let's return early in this case.
+    if (this[p.watchGetters].size === 0) {
+      return Promise.resolve();
+    }
+
+    const selectors = Array.from(this[p.watchGetters].values()).map(
+      ({ id }) => ({ id })
+    );
+
+    return this.put('channels/get', selectors)
+      .then((response) => {
+        Object.keys(response).forEach((key) => {
+          const getter = this[p.watchGetters].get(key);
+          if (!getter) {
+            return;
+          }
+
+          this[p.updateGetterValue](getter, response[key]);
+        });
+      });
+  }
+
+  /**
+   * Updates getter value if needed. If value has changed, appropriate event is
+   * fired.
+   *
+   * @param {{ id: string, value: Object }} getter Getter to update value for.
+   * @param {Object} getterValue Getter value returned from the server.
+   *
+   * @private
+   */
+  [p.updateGetterValue](getter, getterValue) {
+    let valueChanged = false;
+
+    if (!getterValue || !getter.value) {
+      valueChanged = getterValue !== getter.value;
+    } else {
+      const [valueKind] = Object.keys(getterValue);
+      if (valueKind === 'Error') {
+        console.error(
+          'Failed to retrieve value for getter (%s): %o',
+          getter.id,
+          getterValue[valueKind]
+        );
+
+        return;
+      }
+
+      const newValue = getterValue[valueKind];
+      const oldValue = getter.value[valueKind];
+
+      if (newValue && oldValue && typeof newValue === 'object') {
+        // If value is a non-null object, we use their JSON representation to
+        // compare values.
+        valueChanged = JSON.stringify(newValue) !== JSON.stringify(oldValue);
```

Comment:
Definitely, the main reason why I don't really care much - we don't have such values currently as far as I know, so I didn't want to spend time on something that we don't use - just a basic placeholder. If one day we have such values (or when we have more time) - I think we can reuse ```isSimilar``` method from ```services (it will be moved to a common place and maybe improved even more).

What do you think?